extensions/resource-deployment/src/services/notebookService.ts

```diff
@@ -25,18 +24,22 @@ export class NotebookService implements INotebookService {
 	 * Copy the notebook to the user's home directory and launch the notebook from there.
 	 * @param notebook the path of the notebook
 	 */
-	launchNotebook(notebook: string | NotebookInfo): void {
+	launchNotebook(notebook: string | NotebookInfo): Thenable<azdata.nb.NotebookEditor> {
 		const notebookPath = this.getNotebook(notebook);
```

Comment:
This would be a lot cleaner if you made it async and returned a Promise

Reply:
this is calling an API that returns Thenable. I'll leave it as is.